src/tool/pmix_tool.c

```diff
@@ -0,0 +1,691 @@
+    struct sockaddr_un address;
+    pmix_nspace_t *nsptr;
+    pmix_cb_t cb;
+    int errhandler_ref;
+    size_t n;
+
+    if (NULL == proc) {
+        return PMIX_ERR_BAD_PARAM;
+    }
+
+    if (0 < pmix_globals.init_cntr) {
+        /* since we have been called before, the nspace and
+         * rank should be known. So return them here if
+         * requested */
+         if (NULL != proc) {
+            (void)strncpy(proc->nspace, pmix_globals.myid.nspace, PMIX_MAX_NSLEN);
+            proc->rank = pmix_globals.myid.rank;
+        }
+        ++pmix_globals.init_cntr;
+        return PMIX_SUCCESS;
+    }
+
+    /* scan incoming info for directives */
+    if (NULL != info) {
+        for (n=0; n < ninfo; n++) {
+            if (0 == strcmp(PMIX_EVENT_BASE, info[n].key)) {
+                pmix_globals.evbase = (pmix_event_base_t*)info[n].value.data.ptr;
+                pmix_globals.external_evbase = true;
+            }
+        }
+    }
+    /* setup the globals */
+    pmix_globals_init();
+    PMIX_CONSTRUCT(&pmix_client_globals.pending_requests, pmix_list_t);
+    PMIX_CONSTRUCT(&pmix_client_globals.myserver, pmix_peer_t);
+    /* mark that we are a client */
+    pmix_globals.server = false;
+    /* get our effective id's */
+    pmix_globals.uid = geteuid();
+    pmix_globals.gid = getegid();
+    /* initialize the output system */
+    if (!pmix_output_init()) {
+        return PMIX_ERROR;
+    }
+
+    /* default to our internal errhandler */
+    errhandler_ref = 0;
+    pmix_add_errhandler(myerrhandler, NULL, 0, &errhandler_ref);
+    /* see if debug is requested */
+    if (NULL != (evar = getenv("PMIX_DEBUG"))) {
+        debug_level = strtol(evar, NULL, 10);
+        pmix_globals.debug_output = pmix_output_open(NULL);
+        pmix_output_set_verbosity(pmix_globals.debug_output, debug_level);
+    }
+
+    pmix_output_verbose(2, pmix_globals.debug_output,
+                        "pmix: init called");
+
+    nsptr = PMIX_NEW(pmix_nspace_t);
+
+     /* Get the local hostname, and look for a file named
+     * /tmp/pmix.hostname.tool - this file will contain
+     * the URI where the server is listening. The URI consists
+     * of 3 parts - the code below will parse the string read
+     * from the file and connect accordingly */
+ 
+    int i, server_pid = -1;
+    for(i = 0; i < (int)ninfo; i++) {
+      if(strcmp(info[i].key, PMIX_SERVER_PIDINFO) == 0) {
+        server_pid = info[i].value.data.integer;
+      }
+    }
+
+    if(server_pid == -1) {
+      return PMIX_ERR_NOT_FOUND;
+    }
+
+    /* setup the path to the daemon rendezvous point */
+    memset(&address, 0, sizeof(struct sockaddr_un));
+    address.sun_family = AF_UNIX;
+    
+    /* Get first 10 char's of hostname to match what the server is doing */
+    int hostnamelen = 10;
+    char hostname[hostnamelen];
+    gethostname(hostname, hostnamelen);
+
+    snprintf(address.sun_path, sizeof(address.sun_path)-1, "/tmp/pmix.%s.%d", hostname, server_pid);
```

Comment:
I must just be missing it - where did this file get created? I don't see that code.